main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -679,6 +703,30 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       SP ~ "(" ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ ")" ~ SP ~> ParsedAst.Expression.Tuple
     }
 
+    def ArrayLit: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("[|") ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ atomic("|]") ~ SP ~> ParsedAst.Expression.ArrayLit
+    }
+
+    def ArrayNew: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("[|") ~ optWS ~ Expression ~ optWS ~ ";" ~ optWS ~ Expression ~ optWS ~ atomic("|]") ~ SP ~> ParsedAst.Expression.ArrayNew
+    }
+
+    def ArrayLength: Rule1[ParsedAst.Expression] = rule {
+      SP ~ "[" ~ optWS ~ Expression ~ optWS ~ "]" ~ atomic("length") ~ SP ~> ParsedAst.Expression.ArrayLength
+    }
+
+    def VecLit: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("V[") ~ optWS ~ zeroOrMore(Expression).separatedBy(optWS ~ "," ~ optWS) ~ optWS ~ "]" ~ SP ~> ParsedAst.Expression.VecLit
+    }
+
+    def VecNew: Rule1[ParsedAst.Expression] = rule {
+      SP ~ atomic("V[") ~ optWS ~ Expression ~ optWS ~ ";" ~ optWS ~ Literals.Int ~ optWS ~ "]" ~ SP ~> ParsedAst.Expression.VecNew
```

Comment:
This should be a 32bit integer, because java arrays (which we will eventually use) have a length of 32 bit.